### PR TITLE
feat: label un-audited challenges

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
This shows how we could control this at the certificate level.  If you set the `CURRICULUM_LOCALE=espanol` and go to learn/responsive-web-design/ it should not be visible, but it should be on learn/front-end-libraries/

### CODE REVIEW

**Comment by nhcarrigan** [comment]
Do we want this to display on every challenge?

Or on the superblock buttons on the map?

**Comment by raisedadead** [comment]
We would ideally display:

Landing page: No change whatsoever.
/learn: A horizontal separation between challenges ready and those pending translation, with a tiny CTA link to translate.
/learn/super-block: One pill per course (aka block).
/learn/super-block/block/challenge: Somewhere with a tiny CTA link to translate.

**Comment by nhcarrigan** [comment]
image

`/learn` map separated by translation status.

File: client/src/components/Map/index.js [hunk 1]
```diff
@@ -69,6 +71,51 @@ function renderLearnMap(nodes, currentSuperBlock = '') {
...
+              to={`/learn/${dasherize(node.superBlock)}/`}
+            >
+              <div style={linkSpacingStyle}>
+                {generateIconComponent(node.superBlock, 'map-icon')}
+                {createSuperBlockTitle(node.superBlock)}
+              </div>
+            </Link>
+          </li>
+        ))}
+      <hr />
```
Comment (ojeytonwilliams) [inline client/src/components/Map/index.js:93]: With the CTA at the bottom, I don't think it's clear what this `hr` indicates.  Could we put the text here?

For the text itself, how about

"The following certificates are still being translated.  Please go here(link to contribute docs) if you think you can help."

Comment (raisedadead) [reply client/src/components/Map/index.js:93]: I agree, we should move that up.

**Comment by ojeytonwilliams** [comment]
@nhcarrigan I'm converting my bit from `TRANSLATION PENDING` to pills.

File: client/src/components/Map/index.js (hunk 1 above)
Comment (nhcarrigan) [reply client/src/components/Map/index.js:93]: I'll get this moved :)

**Comment by nhcarrigan** [comment]
> @nhcarrigan I'm converting my bit from TRANSLATION PENDING to pills.

Sounds good. I had a draft started for pills, but finally got frustrated enough with Windows and moved my dev environment to WSL. (Without remembering to push up that draft, of course 😅 )

**Comment by ojeytonwilliams** [comment]
Current plan for style of CTAs/warnings:

- /learn/super-block: link to contribution docs (with the words 'help translate').
- /learn/super-block/block/challenge: subtle (info colour) banner (again 'help translate')

**Comment by ojeytonwilliams** [comment]
> Sounds good. I had a draft started for pills, but finally got frustrated enough with Windows and moved my dev environment to WSL. (Without remembering to push up that draft, of course  )

Ouch... I hope you didn't lose too much.

I'll push up those changes ASAP so you can play around with them.  I'll not touch what you've already worked on, just the two paths I mentioned above.

**Comment by ojeytonwilliams** [comment]
image

The 'ES' is just for debugging.

**Comment by nhcarrigan** [comment]
Hmm - I think it'd be better under the title, rather than next to it. 🤔

**Comment by ojeytonwilliams** [comment]
I honestly can't decide.  I've tried above, below and to the right, but none of them looked great.  To the right looked least bad to me, but not by much.

**Comment by nhcarrigan** [comment]
image

Better `/learn` CTA

**Comment by ojeytonwilliams** [comment]
image

How this looks on http://localhost:8000/learn/javascript-algorithms-and-data-structures/

**Comment by nhcarrigan** [comment]
image

Thoughts on rounding it off to look more like a pill?

**Comment by nhcarrigan** [comment]
image

Also wondering if we should set a fixed width

**Comment by ojeytonwilliams** [comment]
I'm not sure about the pills, since the site's aesthetic is quite sharp.  Aside from a few circles (check marks and test icons) it's pretty much all straight lines.

The text shouldn't wrap, though, I agree. I've pushed a commit to stop that element shrinking.

**Comment by nhcarrigan** [comment]
I'll give this a rebase to resolve the conflict.

**Comment by nhcarrigan** [comment]
@ojeytonwilliams I think we've covered everything now - is this ready for review?

**Comment by raisedadead** [comment]
Quick rebase for the QA please?

**Comment by ojeytonwilliams** [comment]
Just resolving that now.

**Comment by raisedadead** [comment]
Here is one more take:

image


image


image

image

**Comment by raisedadead** [comment]
I added a spacer between the CTA on the `/learn`. Let me know if you would like me to add the other styles for the block and title CTA in the screenshot above.

**Comment by nhcarrigan** [comment]
Ooo I really like those. I'm 100% for it.

**Comment by ojeytonwilliams** [comment]
> Let me know if you would like me to add the other styles for the block and title CTA in the screenshot above.

If it's good to go, yeah, push it up.

**Comment by nhcarrigan** [comment]
Hey, with 685e2db included here, landing this should mean we're good to go for the Crowdin uploads again right?

**Comment by raisedadead** [comment]
I edited the issue comments in this PR to collapse the previous screenshots (just to make it easy to catch the current status of the PR).

> Hey, with 685e2db included here, landing this should mean we're good to go for the Crowdin uploads again right?

Yes - after this has been through the QA.

**Review by RandellDawson — DISMISSED** [review]
I tested this locally and it seems to achieve the desired goal.  I kind of wish the labels were all the same height to avoid seeing blue boxes of various heights (due to the text inside the box).  However, these are temporary and will slowly come off as curriculum sections are translated.

**Comment by ahmadabdolsaheb** [comment]
> I tested this locally and it seems to achieve the desired goal. I kind of wish the labels were all the same height to avoid seeing blue boxes of various heights (due to the text inside the box). However, these are temporary and will slowly come off as curriculum sections are translated.
working on those :)

**Comment by ahmadabdolsaheb** [comment]
@ojeytonwilliams great work. @raisedadead great design. I just changed alignment, some margin, and replaced the hover state.

The reason we are doing underline for the breadcrumbs is that @nhcarrigan and I have hid a triangle in there that does not play well with hover. We should probably combine it with the left breadcrumb and overlap the right side so each section can be styled onhover properly.

Finally the highlight colors seem too saturated that they steal the show, I will circle back to them and tone them down a little bit.

LGMT.

**Comment by raisedadead** [comment]
I am going to update the messages that we agreed upon in the chat in a bit and land this.

**Review by raisedadead — DISMISSED** [review]
I would like someone else to approve as well because that last commit is mine.

**Comment by ojeytonwilliams** [comment]
I can review it - it's mostly not my work now.

File: client/src/templates/Introduction/intro.css [hunk 2]
```diff
@@ -8,11 +8,40 @@
...
+  color: var(--highlight-background);
+  background-color: var(--highlight-color);
+}
+
+.block-cta-wrapper {
+  display: flex;
+  flex-direction: column;
+  justify-content: start;
+  padding-right: 15px;
+  padding-top: 25px;
```
Comment (ojeytonwilliams) [inline client/src/templates/Introduction/intro.css:42]:
```suggestion
  justify-content: center;
  padding-right: 15px;
```
@ahmadabdolsaheb couple of things: I got a warning about `start` from Gatsby and don't we just want it to be centered? It looks better to me centered.

**Review by ojeytonwilliams — COMMENTED** [review]
The night mode colours are a bit ugly
<details>

image

image
</details>


something like this looks better to me:

<details>

image

</details>

@ahmadabdolsaheb were you planning on revisiting them, checking contrast and so on?  If so, we can go ahead with this PR.
 
Apparently I can't request changes on my own PR for... reasons, but still:

**Comment by ahmadabdolsaheb** [comment]
Seems like 'misc.translation-pending' and 'learn.help-translate-link' could be combined since both are the same string: "Help us translate.". But it could stay for now.

File: client/src/templates/Introduction/intro.css (hunk 2 above)
Comment (ahmadabdolsaheb) [reply client/src/templates/Introduction/intro.css:42]: Sure, we could either place it in accordance with card or the title. Both are alright with me.

Comment (ahmadabdolsaheb) [reply client/src/templates/Introduction/intro.css]: I will give our Accent Colors a round of audit. That could come in another pr.

File: client/i18n/translations-schema.js [hunk 3]
```diff
@@ -448,7 +448,7 @@ const translationsSchema = {
     email: 'Email',
     and: 'and',
     'change-theme': 'Sign in to change theme.',
-    'translation-pending': 'Help translate'
+    'translation-pending': 'Help us translate.'
```
Comment (ahmadabdolsaheb) [inline client/i18n/translations-schema.js:451]: some translation-pending have . and some don't. do we need the or not?

**Comment by raisedadead** [comment]
> Seems like 'misc.translation-pending' and 'learn.help-translate-link' could be combined since both are the same string: "Help us translate.". But it could stay for now.

Yes, I wanted to consolidate both but kept them separate because one appears with a "stop" symbol on the `/learn` and the other is a button label.

We could re-visit it. 

I think this is good to go.

**Review by ahmaxed — DISMISSED** [review]

File: client/i18n/translations-schema.js (hunk 3 above)
Comment (raisedadead) [reply client/i18n/translations-schema.js:451]: The reason is one is a sentence, as in a link in a sentence. And the other is a button label. We could re-visit this once its out.

File: client/i18n/translations-schema.js [hunk 4]
```diff
@@ -445,7 +447,8 @@ const translationsSchema = {
     'update-email-2': 'Update your email address here:',
     email: 'Email',
     and: 'and',
-    'change-theme': 'Sign in to change theme.'
+    'change-theme': 'Sign in to change theme.',
+    'translation-pending': 'Help us translate.'
```
Comment (raisedadead) [inline client/i18n/translations-schema.js:451]:
```suggestion
    'translation-pending': 'Help us translate'
```

@ahmadabdolsaheb you are correct. Fixing this.

**Review by raisedadead — APPROVED** [review]
For the final time.

**Comment by ojeytonwilliams** [comment]
I can't technically can't approve, but I do approve.

**Pull request merged**